### PR TITLE
Improve code quality

### PR DESCRIPTION
**Pull request by haesleinhuepf**
This is a bulk PR. I'm sychronizing all methods, adding the `plugin_function` annotation (for automatic output-image generation). Furthermore, output images should be returned.
I'm also adding (partly auto-generated) numpy-style documentation.
After this, all methods work as similar as possible.

### CODE REVIEW

File: pyclesperanto_prototype/_tier0/_push.py [hunk 1]
```diff
@@ -3,13 +3,20 @@
 
 
 def push(any_array):
-    """Push numpy array on device to an OpenCL array on GPU.
+    """Copies an image specified by its name to GPU memory in order to process it there later.    Parameters
+    ----------
```
Comment (jni) [inline pyclesperanto_prototype/_tier0/_push.py:7]: Looks like your autogenerator is missing a couple of newlines? =)

```suggestion
    """Copies an image specified by its name to GPU memory in order to process it there later.
    
    Parameters
    ----------
```

Comment (jni) [reply pyclesperanto_prototype/_tier0/_push.py:7]: Also this docstring is missing a Returns section

File: pyclesperanto_prototype/_tier1/_add_image_and_scalar.py [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
+        
     Returns
     -------
+    destination
+
+    Examples
+    --------
+    >>> import pyclesperanto_prototype as cle
+    >>> cle.add_image_and_scalar(source, destination, scalar)
+    >>>
```
Comment (jni) [inline pyclesperanto_prototype/_tier1/_add_image_and_scalar.py:28]:
```suggestion
    >>> destination = cle.add_image_and_scalar(source, destination, scalar)
```

?

File: pyclesperanto_prototype/_tier1/_add_images_weighted.py [hunk 3]
```diff
@@ -4,23 +4,38 @@
 from .._tier0 import plugin_function
 from .._tier0 import Image
 
-@plugin_function(output_creator=create_like)
+@plugin_function
```
Comment (jni) [inline pyclesperanto_prototype/_tier1/_add_images_weighted.py:7]: Has that become the default output creator? Or...?

File: pyclesperanto_prototype/_tier1/_binary_subtract.py [hunk 4]
```diff
@@ -1,19 +1,33 @@
...
-
+    
     Parameters
     ----------
-    (Image minuend, Image subtrahend, ByRef Image destination)
-    todo: Better documentation will follow
-          In the meantime, read more: https://clij.github.io/clij2-docs/reference_binarySubtract
-
-
+    minuend : Image
```
Comment (jni) [inline pyclesperanto_prototype/_tier1/_binary_subtract.py:11]: params don't match actual signature?

File: pyclesperanto_prototype/_tier1/_generate_distance_matrix.py [hunk 5]
```diff
@@ -6,8 +6,31 @@
 @plugin_function(output_creator=create_matrix_from_pointlists)
 def generate_distance_matrix(pointlist1 :Image, pointlist2 :Image, distance_matrix_output :Image = None):
     from .._tier1 import set
-    """
-    docs
+    """Computes the distance between all point coordinates given in two point lists.
+    
+    Takes two images containing pointlists (dimensionality n * d, n: number of points and d: dimensionality) and builds up a matrix containing the distances between these points. 
+    
+    Convention: Given two point lists with dimensionality n * d and m * d, the distance matrix will be of size(n + 1) * (m + 1). The first row and column contain zeros. They represent the distance of the objects to a theoretical background object. In that way, distance matrices are of the same size as touch matrices (see generateTouchMatrix). Thus, one can threshold a distance matrix to generate a touch matrix out of it for drawing meshes.    Parameters
```
Comment (jni) [inline pyclesperanto_prototype/_tier1/_generate_distance_matrix.py:13]: Text should be wrapped at 80c

File: pyclesperanto_prototype/_tier1/_generate_touch_matrix.py [hunk 6]
```diff
@@ -6,8 +6,28 @@
 @plugin_function(output_creator=create_square_matrix_from_labelmap)
 def generate_touch_matrix(labelmap :Image, touch_matrix_output :Image = None):
     from .._tier1 import set
-    """
-    docs
+    """Takes a labelmap with n labels and generates a (n+1)*(n+1) matrix where all pixels are set to 0 exept those where labels are touching.
```
Comment (jni) [inline pyclesperanto_prototype/_tier1/_generate_touch_matrix.py:9]: Cool! This is also called a Region Adjacency Graph, might want to reference that term somehow.

**Comment by jni** [comment]
@haesleinhuepf awesome! I've left a few comments, but then realised the scale of the review, so I'll let you point me to bits you might be unsure about. =)

File: pyclesperanto_prototype/_tier1/_add_images_weighted.py (hunk 3 above)
Comment (haesleinhuepf) [reply pyclesperanto_prototype/_tier1/_add_images_weighted.py:7]: Yes, correct!

File: pyclesperanto_prototype/_tier0/_push.py (hunk 1 above)
Comment (haesleinhuepf) [reply pyclesperanto_prototype/_tier0/_push.py:7]: Drawbacks of auto-generating documentation. 😄

**Comment by haesleinhuepf** [comment]
Alright. Thanks @jni for taking a look! Nothing is perfect, especially if it's auto-generated. :D

I'll merge it as it brings us a step ahead. :-)

**Pull request merged**